src/transforms/hexbin.js

```diff
@@ -0,0 +1,57 @@
+import {hexbin as Hexbin} from "d3-hexbin"; // TODO inline
+import {sqrt4_3} from "../symbols.js";
+import {basic} from "./basic.js";
+import {hasOutput, maybeOutputs} from "./group.js";
+
+export function hexbin(outputs = {fill: "count"}, options = {}) {
+  const {radius, ...rest} = outputs;
+  return hexbinn(rest, {radius, ...options});
+}
+
+// TODO group by (implicit) z
+// TODO filter e.g. to show empty hexbins?
+// TODO data output with sort and reverse?
+// TODO disallow x, x1, x2, y, y1, y2 reducers?
+function hexbinn(outputs, {radius = 10, ...options}) {
+  radius = +radius;
+  outputs = maybeOutputs(outputs, options);
+  return {
+    symbol: "hexagon",
+    ...!hasOutput(outputs, "r") && {r: radius},
+    ...hasOutput(outputs, "fill") && {stroke: "none"},
+    ...basic(options, (data, facets) => {
+      for (const o of outputs) o.initialize(data);
+      return {data, facets};
+    }),
+    initialize(facets, {x: X, y: Y}, {x, y}) {
+      if (X === undefined) throw new Error("missing channel: x");
+      if (Y === undefined) throw new Error("missing channel: y");
+      ({value: X} = X);
```

Comment:
Is this equivalent to saying `X = X.value`?

Reply:
Yes.

Reply:
Perhaps writing it that way would make the code easier to read (unless there's a reason to keep it this way).

Reply:
Maybe? But it’s not like we can avoid destructuring entirely—we use it throughout the codebase. And I don’t think destructuring in general is an antipattern or something that is hard to read.